src/main/java/com/spartronics4915/frc2019/lidar/LidarProcessor.java

```diff
@@ -234,17 +246,22 @@ public void onStart(double timestamp) {
     @Override
     public void onLoop(double timestamp) {
         if (timestamp - getPrevTimestamp() > Constants.kLidarRestartTime) {
-            // XXX: Is this a bug? It seems like this starts and stops the LIDAR server every kLidarRestartTime seconds
+            // XXX: Is this a bug? It seems like this starts and stops the LIDAR server
+            // every kLidarRestartTime seconds
```

Comment:
This really seems like a bug to me... It looks like this code constantly starts and stops the LIDAR server. Am I missing something?

Reply:
@dbadb

Reply:
i see three states:  "Server started", "Server restarting", "Server stopped".  I wonder if we need four: "Server running"... This might be equivalent to "Server started", so not a big deal

Reply:
i agree that there is some clear design that says they want to restart the lidar every so often.  That sounds troubling.

Reply:
assuming IsRunning() is actually a good thing, i wonder if it was supposed to be if(!IsRunning())..

Reply:
`isRunning()` appears to be a good thing... I think we should test this out on a real robot before we change it (they may know something we don't), but I suspect we'll have to change it to `!isRunning()` when the time comes.

Reply:
Even if it was supposed to be `!isRunning()`, this would be buggy... It would work better like this, I think:

```java
           if (!mLidarServer.isEnding() && !mLidarServer.isRunning()) {
                if (mLidarServer.start()) {
                    setPrevTimestamp(timestamp);
                    SmartDashboard.putString("Lidar/status", "Server started");
                } else {
                    SmartDashboard.putString("Lidar/status", "Server couldn't start; sensor " +
                        (mLidarServer.isLidarConnected() ? "is" : "is not") + " connected");
                }
            }
```

Reply:
we agree that !isRunning() is insufficient... Might be good to hide twisty logic within mLidarServer, something like:  mLidarServer.assertRunning()